app/shell2/code.cpp

```diff
@@ -133,7 +135,17 @@ FullCommandList::FullCommandList(EgalitoInterface *egalito) {
             {"-k", ArgumentSpec({"-k"}, ArgumentSpec::TYPE_FLAG)}
         }, {}, 0),
         [this, egalito] (ShellState &state, ArgumentValueList &args) {
-            std::string output = tempnam("/tmp", "ega-");
+            //std::string output = tempnam("/tmp", "ega-");
+            char pfnam[TPATH_MAX];
+            char tfnam[TPATH_MAX];
+
+            memset(pfnam, 0, TPATH_MAX);
+            memset(tfnam, 0, TPATH_MAX);
+
+            snprintf(pfnam, TPATH_MAX-1, "/proc/self/fd/%d",
+                mkstemp((char *)"ega-XXXXXX"));
```

Comment:
Fancy. But, can't use string constant as arg to `mkstemp` as per man page. Also, please use C++-style casts when possible.
```
       The  last six characters of template must be "XXXXXX" and these are re‐
       placed with a string that makes the filename unique.  Since it will  be
       modified,  template  must  not  be a string constant, but should be de‐
       clared as a character array.
```

Reply:
Oops, that's actually wrong -- it's not what I meant to commit 😞.